source/structured_data.cc

```diff
@@ -1178,29 +1191,47 @@ namespace aspect
+                                              const bool parse_time_dependent_parameters)
     {
       Utilities::AsciiDataBase<dim>::parse_parameters(prm,
                                                       subsection_name);
 
       prm.enter_subsection(subsection_name);
       {
-        data_file_time_step             = prm.get_double ("Data file time step");
-        const double first_data_file_model_time      = prm.get_double ("First data file model time");
+        if (parse_time_dependent_parameters == true)
+          {
+            data_file_time_step             = prm.get_double ("Data file time step");
+            const double first_data_file_model_time      = prm.get_double ("First data file model time");
 
-        AssertThrow (first_data_file_model_time == 0.0,
-                     ExcMessage("The `First data file model time' parameter "
-                                "has been deactivated and will be removed in a future release. "
-                                "Do not use this parameter and instead provide data files "
-                                "starting from the model start time."));
+            AssertThrow (first_data_file_model_time == 0.0,
+                         ExcMessage("The `First data file model time' parameter "
+                                    "has been deactivated and will be removed in a future release. "
+                                    "Do not use this parameter and instead provide data files "
+                                    "starting from the model start time."));
 
-        first_data_file_number          = prm.get_integer("First data file number");
-        decreasing_file_order           = prm.get_bool   ("Decreasing file order");
+            first_data_file_number          = prm.get_integer("First data file number");
+            decreasing_file_order           = prm.get_bool   ("Decreasing file order");
 
-        if (this->convert_output_to_years() == true)
+            if (this->convert_output_to_years() == true)
+              {
+                data_file_time_step        *= year_in_seconds;
+              }
+          }
+        else
           {
-            data_file_time_step        *= year_in_seconds;
+            AssertThrow (create_filename (0, 0) == create_filename (1, 0),
+                         ExcMessage("A boundary data file name was used that contained a placeholder for "
+                                    "the current file number, but this AsciiDataBoundary object does not support "
+                                    "time dependent information. Please remove the timestep placeholder."));
           }
+
+        // if filename does not contain a placeholder for timestep, no time dependence
+        // do not issue a warning, the parameter file is specifying exactly one file.
+        if (create_filename (0, 0) == create_filename (1, 0))
+          end_time_dependence (false);
```

Comment:
if the function only changes the bool, I would prefer to get rid of the bool argument and just set ``time_dependent=false;`` here.